src/ray/raylet/scheduling/scheduling_policy.cc

```diff
@@ -63,38 +56,37 @@ int64_t SchedulingPolicy::HybridPolicyWithFilter(
     return !has_gpu;
   };
 
-  const auto &local_node_view = local_it->second.GetLocalView();
-  // If we should include local node at all, make sure it is at the front of the list
-  // so that
-  // 1. It's first in traversal order.
-  // 2. It's easy to avoid sorting it.
-  if (predicate(local_node_id_, local_node_view) && !force_spillback) {
-    round.push_back(local_node_id_);
-  }
+  // Step 1: Generate the traversal order. We guarantee that the first node is local, to
+  // encourage local scheduling. The rest of the traversal order should be globally
+  // consistent, to encourage using "warm" workers.
+  std::vector<int64_t> round;
+  round.reserve(nodes_.size());
+  round.emplace_back(local_node_id_);
 
-  const auto start_index = round.size();
   for (const auto &pair : nodes_) {
-    if (pair.first != local_node_id_ &&
-        predicate(pair.first, pair.second.GetLocalView())) {
-      round.push_back(pair.first);
+    if (pair.first != local_node_id_) {
+      round.emplace_back(pair.first);
```

Comment:
Yea, we can also maintain a sorted list of nodes instead of resorting every time. I'll leave this out for a future optimization.

Reply:
Yep as I mentioned in [this comment](https://github.com/ray-project/ray/pull/19968#discussion_r769206622), a standard sorted container would have a few benefits but isn't worth the complexity at this point IMO. Not a high priority since this is most likely not even in the top 3 scalability/perf bottlenecks for our scalability envelope, but we should make sure that our large-cluster scalability test measures this scheduling slowdown so we can easily pinpoint it once it _does_ become the primary bottleneck.